### ISSUE TITLE
transactions hash doesn't match unspent txid

### ISSUE THREAD

**Issue by Abd0s** [ofek/bit#98]
Using the btc testnet I made 2 transactions to my key address. When I then run `get_transactions()` I get 2 transaction hashes back.

`['c3b66dd4951a460def4891053f1baf2e0119a78569db9a756a8eac08d5e7ef62', '52d8b28675b25957247a7ba795fb20ef5fbb9013ac32890692843395de7c3579']`

When I try to look up these hashes I get no result.

If I run get_unspents() I get back.

`[Unspent(amount=1000000, confirmations=1, script='76a9146fec8fac08f8da7be14f5b06b7cc361fc60c192288ac', txid='f673f7d6ccaafaebc16942832ae7f893e63c47f7ce98c0f2dd83eebd30f80150', txindex=1, segwit=False), Unspent(amount=1791776, confirmations=3, script='76a9146fec8fac08f8da7be14f5b06b7cc361fc60c192288ac', txid='7548ca9ce5ce2006dbee3414a1184faf04ef7fee9936f2d92947b55e4d46ae98', txindex=0, segwit=False)]`

Here u can see the txid which do work. These txid differ from the transaction hashes.

When looking at the example in the docs it shows that the txid and transaction hashes should be the same. Am I missing something or is this not working like it should?

**Comment by bjarnemagnussen** [comment]
Hi @Abd0s,

Are you using a Segwit address and a Bit version different from the master branch, specifically with commit a17bff69d83e5f026acba0b1763ccef87d01cd2a checked out?

Here is an issue that has been fixed but could explain the error you see:
https://github.com/ofek/bit/issues/93

Let me know if this fix does not work for you.

**Comment by Abd0s** [comment]
Hi

I'm was not using a Segwit address and I'm using the master branch (`pip install bit`).

I've look through the issue you referenced, but I cant get much out of it.

I've tried a few more transaction and one of it's (not segwit address) txid do match with its hash but other don't. I've alse tried the segwit address, there txid do not match there hash (as expected).

`['be19c6905d80b88e0d9387bbb63f5fd3406bbe02eba9a85abbf40af7259abd91', 'c3b66dd4951a460def4891053f1baf2e0119a78569db9a756a8eac08d5e7ef62', '52d8b28675b25957247a7ba795fb20ef5fbb9013ac32890692843395de7c3579', 'ccfd769b07ee2f1094e4efcae44963123d9ecca4dca1601fb1e491a5c205064e', 'dcfec25964a6fe13a17c07922fef34243eeb2df072053f942cc02fc5a87e44e2']`

**Does match with hash - no segwit**
Unspent(amount=1000000, confirmations=2, script='76a9146fec8fac08f8da7be14f5b06b7cc361fc60c192288ac', txid='be19c6905d80b88e0d9387bbb63f5fd3406bbe02eba9a85abbf40af7259abd91', txindex=0, segwit=False), 

**Does not match with hash - no segwit**
Unspent(amount=1000000, confirmations=61, script='76a9146fec8fac08f8da7be14f5b06b7cc361fc60c192288ac', txid='f673f7d6ccaafaebc16942832ae7f893e63c47f7ce98c0f2dd83eebd30f80150', txindex=1, segwit=False), 
Unspent(amount=1791776, confirmations=63, script='76a9146fec8fac08f8da7be14f5b06b7cc361fc60c192288ac', txid='7548ca9ce5ce2006dbee3414a1184faf04ef7fee9936f2d92947b55e4d46ae98', txindex=0, segwit=False), 

**segwit**
Unspent(amount=1000000, confirmations=0, script='a914c3ccefebab2ae12ecdc8bb115cc4c21c8505f00e87', txid='f6a02cf177823e62de8b0ac5ecf9b1f34921270fcfba496691e388fd40fbd358', txindex=0, segwit=True), Unspent(amount=10000, confirmations=1, script='a914c3ccefebab2ae12ecdc8bb115cc4c21c8505f00e87', txid='22fbbb66822687c3ad6d24ad71592b90d5b7f1bfe2b32c52ae6713f61a2a5222', txindex=1, segwit=True)

I'm not sure whats going wrong here?

**Comment by bjarnemagnussen** [comment]
I have looked more deeply into this and the error is unfortunately the same as the issue I was linking to.

TLDR: The bugfix has not yet been updated in pip, which is why you did still experience the bug unfortunately.

The reason is as follows:
Some of the transactions you received bitcoins with have been using Segwit, and therefore the receiving transactions can be in a Segwit format. For Segwit format `transaction-id != transaction-hash`. The bug is that the method `get_transactions` polls the transaction hashes, whereas it should get ids like e.g. the `get_unspents` method.

You can e.g. take a look at this API call https://testnet-api.smartbit.com.au/v1/blockchain/tx/f673f7d6ccaafaebc16942832ae7f893e63c47f7ce98c0f2dd83eebd30f80150, which shows both the id and hash.

Will try to push the bugfix to pip!

**Comment by bjarnemagnussen** [comment]
Should be fixed with version 0.6.1 of Bit on pypi (https://pypi.org/project/bit/)
